django/contrib/postgres/aggregates/statistics.py

```diff
@@ -0,0 +1,118 @@
+from django.db.models import F, FloatField, IntegerField
+from django.db.models.aggregates import Aggregate
+from django.db.models.expressions import Value
+from django.utils import six
+
+__all__ = [
+    'CovarPop', 'Corr', 'RegrAvgX', 'RegrAvgY', 'RegrCount', 'RegrIntercept',
+    'RegrR2', 'RegrSlope', 'RegrSXX', 'RegrSXY', 'RegrSYY', 'StatFunc',
+]
+
+
+class StatFunc(Aggregate):
+    template = "%(function)s(%(y)s, %(x)s)"
+    _num_expression_alias = 'num'
+
+    def __init__(self, y, x, output_field=FloatField()):
+        if not x or not y:
+            raise TypeError('Both X and Y must be provided.')
+        super(StatFunc, self).__init__(y=y, x=x, output_field=output_field)
+        self.x = x
+        self.y = y
+        self.source_expressions = self._parse_expressions(self.x, self.y)
+
+    def _parse_expressions(self, *expressions):
+        # Some stat functions allows integer to be an argument,
+        # so we need to parse it and not resolve expression as F for
+        # this case.
+        return [
+            F(arg) if isinstance(arg, (six.string_types, six.text_type)) else Value(arg)
+            for arg in expressions
+        ]
+
+    @property
+    def default_alias(self):
+        # Since number is allowed to be an expression,
+        # we need to have kinda "static" alias for this case.
+        x = self._num_expression_alias if not isinstance(self.x, (six.string_types, six.text_type)) else self.x
+        y = self._num_expression_alias if not isinstance(self.y, (six.string_types, six.text_type)) else self.y
+        return '%s_%s__%s' % (y, x, self.name.lower())
```

Comment:
```
.aggregate(regr=RegrSYY('col_a', 'col_b'))
# vs
.aggregate(RegrSYY('col_a', 'col_b')) -> col_a_col_b_regrsyy
```

It's a long and unwieldy default alias. You also have to be careful about the column name length in this situation. Oracle only supports names up to 30 chars, which means both column names need to be 10 chars or less to account for the extra bits of the underscores and aggregate name. I don't see the value.